libs/src/services/solana/SolanaWeb3Manager.ts

```diff
@@ -479,45 +479,47 @@ export class SolanaWeb3Manager {
   }: {
     id: number
     type: 'track'
-    splits: Array<{ address: string; amount: BN }>
+    splits: Record<string, number | BN>
     blocknumber: number
   }) {
     if (!this.web3Manager) {
       throw new Error(
         'A web3Manager is required for this solanaWeb3Manager method'
       )
     }
-    if (splits.length !== 1) {
+    if (Object.values(splits).length !== 1) {
       throw new Error(
         'Purchasing content only supports a single split. Specifying more splits coming soon!'
       )
     }
+
+    const totalAmount = Object.values(splits).reduce<BN>(
+      (sum, split) => (split instanceof BN ? sum.add(split) : sum.addn(split)),
+      new BN(0)
+    )
+
```

Comment:
Why are we supporting both bn and number?

Reply:
When getting the JSON response back from DN, and then passing it back through, the type is number the whole way.

FWIW, I think number is sufficiently large: 9,007,199,254,740,991

Would you say just don't support BN at all?